go/pkg/client/client_context.go

```diff
@@ -19,55 +19,80 @@ const (
 	remoteHeadersKey = "build.bazel.remote.execution.v2.requestmetadata-bin"
 )
 
+// ContextMetadata is optionally attached to RPC requests.
+type ContextMetadata struct {
```

Comment:
Can we just use `RequestMetadata`[1] instead? Or make it a field of `ContextMetadata`?

1: https://github.com/bazelbuild/remote-apis/blob/master/build/bazel/remote/execution/v2/remote_execution.proto#L1677

Reply:
I didn't want to do that specifically, that's why I had the internal struct. I wanted to decouple this from the API protos. Besides, the RequestMetadata proto is clunkier to use (with the internal structure that it has.)

Reply:
Sgtm!